lib/usd/translators/shading/mtlxBaseWriter.cpp

```diff
@@ -117,59 +99,44 @@ UsdPrim MtlxUsd_BaseWriter::GetNodeGraph()
     return UsdShadeNodeGraph::Define(GetUsdStage(), ngPath).GetPrim();
 }
 
-UsdAttribute MtlxUsd_BaseWriter::AddConversion(
-    const SdfValueTypeName& fromType,
-    const SdfValueTypeName& toType,
-    UsdAttribute            nodeOutput)
+UsdAttribute MtlxUsd_BaseWriter::ExtractChannel(size_t channelIndex, UsdAttribute nodeOutput)
```

Comment:
Previous code was `color` centric. This new code will work with vector types as well.